docker/autoware-universe/Dockerfile

```diff
@@ -58,14 +58,70 @@ RUN mkdir -p /etc/OpenCL/vendors \
 RUN echo "source /opt/ros/${ROS_DISTRO}/setup.bash" > /etc/bash.bashrc
 CMD ["/bin/bash"]
 
-FROM devel as prebuilt
+FROM devel as builder
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
 ## Build and change permission for runtime data conversion
 RUN source /opt/ros/"$ROS_DISTRO"/setup.bash \
   && colcon build --cmake-args -DCMAKE_BUILD_TYPE=Release \
   && find /autoware/install -type d -exec chmod 777 {} \;
 
+ARG CUDA_BASE_IMAGE
+# hadolint ignore=DL3006
+FROM $CUDA_BASE_IMAGE as prebuilt
+
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+
+ARG ROS_DISTRO
+ARG SETUP_ARGS
```

Comment:
[Question] Do you need to repeat these args? Also written in the line 6-7.

Reply:
Yes，we have to declare those ARGs in each stage in which want to use the argument. From the docker documentation，“[An ARG instruction goes out of scope at the end of the build stage where it was defined. To use an argument in multiple stages, each stage must include the ARG instruction.](https://docs.docker.com/engine/reference/builder/#:~:text=An%20ARG%20instruction%20goes%20out%20of%20scope%20at%20the%20end%20of%20the%20build%20stage%20where%20it%20was%20defined.%20To%20use%20an%20argument%20in%20multiple%20stages%2C%20each%20stage%20must%20include%20the%20ARG%20instruction.)”

Reply:
Thank you for the information! :smile: